cli/conf.py

```diff
@@ -75,15 +76,34 @@ def parse_args(cls, module_name, config, args=None):
         :param module_name: the module name: installer|provisioner|tester
         """
         cmd = clg.CommandLine(cls._get_specs(module_name, config))
-        res_args = cmd.parse(args)
-
-        # override the command specific arguments from file
+        res_args = vars(cmd.parse(args))
+
+        # always load default values for command0
+        default_file = os.path.join(
+            config.get('defaults', 'settings'),
+            module_name,
+            res_args['command0'],
+            DEFAULT_INI
+        )
+        defaults = IniFileType(default_file)[res_args['command0']]
+
+        # override defaults with env variables
+        for arg_name, arg_value in res_args.iteritems():
+            upper_arg_name = arg_name.upper()
+            if upper_arg_name in os.environ:
```

Comment:
If value was provided in CLI, we ignore everything:
`if arg_value is not None and ...`